app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_post);
 
-        // Configure window settings for fullscreen mode
-        requestWindowFeature(Window.FEATURE_NO_TITLE);
-        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
-                WindowManager.LayoutParams.FLAG_FULLSCREEN);
-        getSupportActionBar().hide();
+        DebugUtils.logUserFoldersAndPosts(User.getUser());
 
-        // Set the activity layout
-        setContentView(R.layout.activity_edit_post);
+        EditText linkToPost = findViewById(R.id.editTextLinkToPost);
+        EditText postTitle = findViewById(R.id.editTextPostTitle);
+        EditText postDescription = findViewById(R.id.editTextPostDescription);
+
+        Button buttonSaveChanges = findViewById(R.id.buttonSaveChanges);
+        Button buttonCancelChanges = findViewById(R.id.buttonCancelChanges);
+
+        String postId = getIntent().getStringExtra("postID");
+        String folderId = getIntent().getStringExtra("folderID");
 
-        // Prep vars
-        String currentFolderTitle;
         Post currentPost;
-        Folder currentFolder;
-        HashMap<Folder, FolderAccess> userFolders;
-        HashMap<String, Integer> folderIdNameMap = new HashMap<>();
-        EditText postNameField = findViewById(R.id.editTextNewName);
-        EditText postLinkField = findViewById(R.id.editTextNewLink);
-        Spinner spinner = findViewById(R.id.edit_post_spinner);
-        Button cancelBtn = findViewById(R.id.edit_post_cancel_btn);
-        Button saveBtn = findViewById(R.id.edit_post_save_button);
-        try {
-            currentPost = API.getPostById(getIntent().getIntExtra("postId", 0));
-            currentFolder = API.getFolderById(getIntent().getIntExtra("folderId", 0));
-            currentFolderTitle = String.format("(%d) %s", currentFolder.getId(), currentFolder.getName());
-            userFolders = API.getFoldersForUserId(PosteApplication.getCurrentUser().getId());
-            for (Folder folder: userFolders.keySet()) {
-                folderIdNameMap.put(String.format("(%d) %s", folder.getId(), folder.getName()), folder.getId());
+        Folder currentFolder = User.getUser().getFolder(folderId);
+        if (currentFolder == null){
+            currentPost = null;
```

Comment:
I like the null check here! You might consider also checking if getFolder returns null; otherwise, a NullPointerException will crash the app here

Reply:
currentFolder == null checks if getFolder returns null, since currentFolder gets the return value assigned to it from getFolder...
Unless I'm missing something?

Reply:
Sorry, I think I was unclear.

In line 54, we getStringExtra(); what if this value isn't passed? Is it possible to be null? Or rather, what happens if we pass an invalid value to `getFolder`?

Reply:
OHH yes that makes sense. I'll make a note of it to get to that ASAP.